tgui/packages/tgui/interfaces/DnaConsole.js

```diff
@@ -653,6 +653,14 @@ const MutationInfo = (props, context) => {
                 is_activator: 0,
                 source: mutation.Source,
               })} />
+            <Button
+              icon="syringe"
+              disabled={!mutation.Active}
+              content="CRISPR"
+              onClick={() => act('crispr', {
+                mutref: mutation.ByondRef,
+                source: mutation.Source,
+              })} />
```

Comment:
Should also be something somewhere on the interface indicating how many CRISPR charges are available.

Reply:
This will probably help with the other code review regarding disabling the CRISPR button.

In `/obj/machinery/computer/scan_consolenew/ui_data(mob/user)`, `var/list/data` is an associated list of data that is sent to the tgui interface.

You can set a value in the data list probably below `data["isScannerConnected"] = can_use_scanner` sending the number of CRISPR charges to tgui, which you can then grab and use as necessary.

If you have any questions, hit up the #coding-general on the /tg/Station13 discord. There's a few tgui wizards who can help you.

Reply:
>Not sure how to check for that in TGUI - if you had some resources I could look over I'm happy to do my own reading otherwise I'll tinker tonight to figure it out

See above. Also https://github.com/tgstation/tgstation/blob/master/tgui/README.md and https://github.com/tgstation/tgstation/blob/master/tgui/docs/tutorial-and-examples.md